forum/backends/mongodb/users.py

```diff
@@ -111,3 +111,22 @@ def update(
             {"$set": update_data},
         )
         return result.modified_count
+
+    def delete_read_state_by_thread_id(self, thread_id: str) -> None:
+        """Delete read state from users based on thread_id."""
+        users = self.get_list(
+            **{
+                "read_states.last_read_times": {"$exists": True},
+                f"read_states.last_read_times.{thread_id}": {"$exists": True},
+            }
+        )
+        for user in list(users):
+            updated_read_states = []
+            for read_state in user.get("read_states", []):
+                if "last_read_times" in read_state:
+                    if thread_id in read_state["last_read_times"]:
```

Comment:
These two "if" checks shouldn't be necessary according to the mongodb query that we are making right?

Reply:
yup, we can delete first if condition. But we can't delete the second if condition, because there can be last_read_times for other thread_ids as well and we only want to delete if thread_id matches.

Reply:
yes, you are right :) let me make a change!